### PR TITLE
Add feature: override role name displayed to end user from config option

### PR DESCRIPTION
**Pull request by Kirrus-Indiestone**
This patch adds a configuration option which will let end users arbitrarily override the role displayed to end users from the config option 
`overrideRoleNameDisplay`

This allows to hide a moderator with a hitched role which is unsuitable to be displayed to end users. In our case, specifically a hidden moderator with a general user role. Not able to use `fallbackRoleName`, because the hidden moderator has a hitched role.

I have not coded in a very long time, and never in JS before, so rusty doesn't even begin to cover it. Thoughts/rejection/comments/patches appreciated :)

I've tested this patch on our server, and it appears to work okay, but have not tested all existing functions exhaustively

### CODE REVIEW

**Comment by Kirrus-Indiestone** [comment]
TODO: 
- Verify fallback works when override not in effect
- Document option in configuration.md

**Comment by Kirrus-Indiestone** [comment]
Fallback still works whilst override not enabled, configuration option documented

File: docs/configuration.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ To do that, use the same format as with "Accepts multiple values" above:
 ```ini
 greetingMessage[] = Welcome to the server!
 greetingMessage[] = This is the second line of the greeting.
-greetingMessage[] = 
+greetingMessage[] =
```
Comment (DarkView) [inline docs/configuration.md:59]: Unnecessary formatting change

File: docs/configuration.md [hunk 2]
```diff
@@ -300,6 +300,13 @@ If enabled, a system message will be posted into any open threads if the user jo
 **Default:** `on`  
 If enabled, a system message will be posted into any open threads if the user leaves a main server
 
+#### overrideRoleNameDisplay
+**Default:** `None`  
+Role name to display in all replies. This completely overrides normal role selection, all replies will contain the string entered. For example;
+```ini
```
Comment (DarkView) [inline docs/configuration.md:306]: Full code block comments should be reserved for more elaborate config options.
I would change this to use regular inline highlights using ticks, like here for example:

> #### reactOnSeenEmoji
> **Default:** `📨`  
> The emoji that the bot will react with when it sees a message.  Requires `reactOnSeen` to be enabled.  
> Must be pasted in the config file as the Emoji representation and not as a unicode codepoint. Use `emojiName:emojiID` for custom emoji.

File: docs/configuration.md (hunk 1 above)
Comment (Kirrus-Indiestone) [reply docs/configuration.md]: Fixed -- sorry, my editor,  Atom doesn't like hanging spaces. Added back in to revert the formatting change

File: docs/configuration.md (hunk 2 above)
Comment (Kirrus-Indiestone) [reply docs/configuration.md]: Done

**Comment by Dragory** [comment]
This looks good, thank you for the PR!
